### PR TITLE
Ember Data ES6 macro test

### LINKED ISSUE #98: Ember Data decorators do not appear to work with ES6 classes
```
import DS from 'ember-data';
import { attr } from 'ember-computed-decorators/ember-data';

export default class extends DS.Model {
  @attr title;
};
```

Using Babel 6 with this EmberApp config:
```
    babel: {
      plugins: [
        'transform-decorators-legacy',
        "transform-class-properties"
      ]
    }
```

### PR DESCRIPTION
**Pull request by snewcomer**
Maybe I'm missing something related to the below issue or just magically works (:man_shrugging:), but here is a passing test.

Fix #98

### CODE REVIEW

**Comment by buschtoens** [comment]
#98 is about using ES6 class syntax instead of the Ember Object Model, i.e. this

```js
import Model from 'ember-data/model';
import { attr } from 'ember-computed-decorators/ember-data';

export default class Post extends Model {
  @attr('string') title;
}
```

instead of this

```js
import Model from 'ember-data/model';
import { attr } from 'ember-computed-decorators/ember-data';

export default Model.extend({
  @attr('string') title: null
});
```

Your PR adopts [RFC #176](https://github.com/ember-cli/ember-rfc176-data) style imports. :blush:

**Comment by snewcomer** [comment]
@buschtoens oh I think I added a test at the bottom for the ES6 class test..

**Comment by buschtoens** [comment]
Didn't read that carefully enough. Sorry!

Very curious that it seems to work now. I could _swear_ that it did not a few weeks ago.

**Comment by pzuraq** [comment]
I’d like to set up some tests with full Models that have two sided relatioships, really put it through its paces. I have a feeling that the way we are testing here is not triggering whatever the hugs are with the decorators

File: tests/unit/ds-macro-test.js [hunk 1]
```diff
@@ -35,10 +35,101 @@ test('DS macro', function(assert) {
...
+
+  let user_relationships = [];
+  User.eachRelationship(function(attr, meta) {
+    user_relationships.push({key: attr, type: meta.type});
+  });
+
+  assert.deepEqual(user_relationships, [
+    {key: 'storefronts', type: 'storefront'}
+  ]);
+});
```
Comment (snewcomer) [inline tests/unit/ds-macro-test.js]: Next, I could take a shot at registering models with the registry, similar to [this](https://github.com/emberjs/data/blob/master/tests/helpers/store.js) and then do a store.push...then store.find to ensure the relationships are setup correct. Thoughts?

File: tests/unit/ds-macro-test.js [hunk 2]
```diff
@@ -1,44 +1,236 @@
...
+    });
+  });
+
+  return run(() => {
+    const tag = store.peekRecord('tag', 5);
+    assert.equal(get(tag, 'name'), 'friendly', 'retrieves tag record from store');
+    const user = store.peekRecord('user', 1);
+    assert.equal(get(user, 'name'), 'Tom Dale', 'retrieves user record from store');
+    // assert.equal(get(user, 'tag') instanceof Tag, true, 'the tag property should return a tag');
+    // assert.equal(get(user, 'tag.name'), 'friendly', 'the tag should have name');
```
Comment (snewcomer) [inline tests/unit/ds-macro-test.js]: @buschtoens @pzuraq Ah here we go.  So `parentType` is not defined for the relationship model (ED [line](https://github.com/emberjs/data/blob/0999f8a0a756a3118f1e6a4e110709be72e1ad3e/addon/-private/system/model/model.js#L1323)).  Maybe ya'll already knew this :)  See the test above that shows pushing/finding the relationship with object protos work whereas this test doesn't.

Comment (pzuraq) [reply tests/unit/ds-macro-test.js]: Sweet, thanks for tracking this down! Will help a lot with figuring out a fix. I should be able to get to this sometime after the holidays

**Comment by pzuraq** [comment]
This was superceded by the work to breakout @ember-decorators/data

**Pull request closed without merging**